mne/preprocessing/eyetracking/calibration.py

```diff
@@ -0,0 +1,346 @@
+        assert structured_array.ndim == 1
+        if not len(structured_array[0]) == 5:
+            raise ValueError(
+                f"Each column in data must have have 5 elements: got {data}"
+            )
+        self["points"] = structured_array
+
+    def plot(self, title=None, show_offsets=False, invert_y_axis=True, show=True):
+        """Visualize calibration.
+
+        Parameters
+        ----------
+        title : str
+            The title to be displayed. Defaults to ``None``, which uses a generic title.
+        show_offsets : bool
+            Whether to display the offset (in visual degrees) of each calibration
+            point or not. Defaults to ``False``.
+        invert_y_axis : bool
+            Whether to invert the y-axis or not. In many monitors, pixel coordinate
+            (0,0), which is often referred to as origin, is at the top left of corner
+            of the screen. Defaults to ``True``.
+        show : bool
+            Whether to show the figure or not. Defaults to ``True``.
+
+        Returns
+        -------
+        fig : instance of matplotlib.figure.Figure
+            The resulting figure object for the calibration plot.
+        """
+        import matplotlib.pyplot as plt
+
+        if not isinstance(self["points"], np.ndarray):
+            raise TypeError(
+                "Calibration points must be a numpy array. Use "
+                "set_calibration_array() to set calibration data."
+            )
+        fig, ax = plt.subplots()
+        px, py = self["points"]["point_x"], self["points"]["point_y"]
+        gaze_x, gaze_y = self["points"]["gaze_x"], self["points"]["gaze_y"]
+
+        if title is None:
+            ax.set_title(f"Calibration ({self['eye']} eye)")
+        else:
+            ax.set_title(title)
+        ax.set_xlabel("x (pixels)")
+        ax.set_ylabel("y (pixels)")
+
+        # Display avg_error and max_error in the top left corner
+        text = (
+            f"avg_error: {self['avg_error']} deg.\nmax_error: {self['max_error']} deg."
+        )
+        ax.text(
+            0,
+            1.01,
+            text,
+            transform=ax.transAxes,
+            verticalalignment="baseline",
+            fontsize=8,
+        )
+
+        if invert_y_axis:
+            # Invert the y-axis because origin is at the top left corner for most
+            # monitors
+            ax.invert_yaxis()
+        ax.scatter(px, py, color="gray")
+        ax.scatter(gaze_x, gaze_y, color="red", alpha=0.5)
+
+        if show_offsets:
+            for i in range(len(px)):
+                x_offset = 0.01 * gaze_x[i]  # 1% to the right of the gazepoint
+                text = ax.text(
+                    x=gaze_x[i] + x_offset,
+                    y=gaze_y[i],
+                    s=self["points"]["offset"][i],
+                    fontsize=8,
+                    ha="left",
+                    va="center",
+                )
+
+        fig.tight_layout()
```

Comment:
Rather than tight layout I'd prefer to create the figure with `constrained_layout=True`, it almost always works better
```suggestion
```